instructions-for-devs.md

```diff
@@ -0,0 +1,199 @@
+xucli <command>
+
+Commands:
+  xucli cancelorder <pair_id> <order_id>    cancel an order
+  xucli connect <host> [port]               connect to an xu node
+  xucli disconnect <host> [port]            disconnect from an xu node
+  xucli executeSwap <identifier> <role>     execute an atomic swap
+  <sending_amount> <sending_token>
+  <receiving_amount> <receiving_token>
+  xucli getinfo                             get general info from the xud node
+  xucli getorders [pair_id] [max_results]   get orders from the order book
+  xucli getpairs                            get order book's available pairs
+  xucli placeorder <pair_id> <order_id>     place an order
+  <quantity> [price]
+  xucli shutdown                            gracefully shutdown the xud node
+  xucli subscribepeerorders                 subscribe to incoming peer orders
+  xucli subscribeswaps                      subscribe to executed swaps
+
+Options:
+  --help          Show help                                            [boolean]
+  --version       Show version number                                  [boolean]
+  --rpc.port, -p  RPC service port                      [number] [default: 8886]
+  --rpc.host, -h  RPC service hostname           [string] [default: "localhost"]
+
+
+```
+
+Examples:
+```bash
+./xucli connect xud1.test.exchangeunion.com 8885
+# Manually connect to another xud instance
+
+./xucli placeorder BTC/LTC 1337 1 99
+#Places a new limit order BUYING 1 BTC for 99 LTC, assigning the orderId 1337
+
+./xucli placeorder BTC/LTC 1338 -1 99
+#Places a new limit order SELLING 1 BTC for 99 LTC, assigning the orderId 1338
+
+./xucli placeorder BTC/LTC 1339 1
+#Places a new market order BUYING 1 BTC for the best LTC market price, assigning the orderId 1339
+
+```
+
+## Configuration (optional)
+
+This *optional* configuration file uses [TOML](https://github.com/toml-lang/toml) and by default should be saved at  `~/.xud/xud.conf` on Linux or `AppData\Local\Xud\xud.conf` on Windows (run `xud` at least once for this folder to be created). Default settings which can be overridden are shown below.
+
+```toml
+[rpc]
+port = 8886
+
+[webproxy]
+disable = false
+port = 8080
+
+[db]
+username = "xud"
+password = ""
+database = "xud"
+port = 3306
+host = "localhost"
+
+[p2p]
+listen = true
+port = 8885
+#make sure this port is reachable from the internet
+
+[lnd]
+disable = false
+host = "localhost"
+
+[raiden]
+disable = false
+host = "localhost"
+port = 5001
+```
+
+## Useful stuff for developers (optional)
+
+### Auto-restart `xud` on file change
+
+Auto restart on every file change under `dist` folder with `nodemon`:
+ ```
+nodemon --watch dist -e js bin/xud
+```
+ With args:
+ ```
+nodemon --watch dist -e js bin/xud --lnd.disable=true
+```
+### Daemonize `xud`
+If you want to daemonize `xud`, so you don't have to keep a terminal open and xud starts on bootup, you can do this for example on Ubuntu by adding the following to `systemd`:
+
+```bash
+[Unit]
```

Comment:
This service file doesn't work properly. My bad. @itsbalamurali did come up with a better solution:
```
[Unit]
Description= XUD - ExchangeUnion Daemon
ConditionPathExists=/opt/xud/dist/
After=network.target

[Service]
User=xud
Group=xud
WorkingDirectory=/opt/xud/
ExecStart=/usr/bin/node /opt/xud/dist/Xud.js
Type=simple

[Install]
WantedBy=multi-user.target
```

Reply:
Swaped it. Thx for clearing that up!